src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -1419,59 +1419,74 @@ public override void Initialize(RachisConsensus parent, TransactionOperationCont
 
         private void OnStateChange(object sender, RachisConsensus.StateTransition transition)
         {
-            if (transition.From == RachisState.Passive && transition.To == RachisState.LeaderElect)
+            if (transition.To != RachisState.LeaderElect)
```

Comment:
This happens as soon as we move to `Passive` state.
But previously, that was when we moved from `Passive` to something else.

How do you handle removal of node from cluster and then adding it again in this case?